context/app/static/js/components/workspaces/utils.js

```diff
@@ -21,33 +21,36 @@ async function createEmptyWorkspace({ workspacesEndpoint, workspacesToken, works
 }
 
 async function stopJob({ jobId, workspacesEndpoint, workspacesToken }) {
-  await fetch(`${workspacesEndpoint}/jobs/${jobId}/stop/`, {
+  const response = await fetch(`${workspacesEndpoint}/jobs/${jobId}/stop/`, {
     method: 'PUT',
     headers: getWorkspacesApiHeaders(workspacesToken),
   });
+  if (!response.ok) {
+    throw Error(`Job stop for job #${jobId} failed`);
+  }
 }
 
 async function deleteWorkspace({ workspaceId, workspacesEndpoint, workspacesToken }) {
+  const headers = getWorkspacesApiHeaders(workspacesToken);
+  const response = await fetch(`${workspacesEndpoint}/workspaces/${workspaceId}/`, {
+    method: 'DELETE',
+    headers,
+  });
+  if (!response.ok) {
+    throw Error(`Workspace deletion for workspace #${workspaceId} failed`);
+  }
+}
+
+async function stopJobs({ workspaceId, workspacesEndpoint, workspacesToken }) {
   const headers = getWorkspacesApiHeaders(workspacesToken);
   const jobsResponse = await fetch(`${workspacesEndpoint}/jobs`, { headers });
   const jobsResults = await jobsResponse.json();
   const { jobs } = jobsResults.data;
   jobs.forEach((job) => {
-    if (String(job.workspace_id) === workspaceId) {
+    if (String(job.workspace_id) === String(workspaceId)) {
```

Comment:
Maybe the API was returning strings at some point? In any event, the types are mismatched right now, and this will be safest.